File: src/main/java/com/example/estantevirtual/repository/BookRepository.java
package com.example.estantevirtual.repository;

import com.example.estantevirtual.model.Book;
import lombok.NonNull;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface BookRepository extends JpaRepository<Book, Long> {

    @Override
    @NonNull Page<Book> findAll(@NonNull Pageable pageable);

    Optional<Book> findById(UUID id);

    @Transactional
    void deleteById(UUID id);
}

File: src/main/kotlin/com/estantevirtual/repository/BookRepository.kt
package com.estantevirtual.repository

import com.estantevirtual.model.Book
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Repository
interface BookRepository : CrudRepository<Book, String> {
    @Query(
        value = "SELECT CONCAT(isbn) as id, * FROM Book",
        countQuery = "SELECT count(*) FROM Book",
        nativeQuery = true
    )
    fun findAll(pageable: Pageable): Page<Book?>

    fun findByIsbn(isbn: String?): Optional<Book?>

    @Transactional
    fun deleteById(id: String?)
}
